Sources/Purchasing/Purchases/PurchasesType.swift

```diff
@@ -1008,6 +1009,35 @@ public protocol PurchasesSwiftType: AnyObject {
 
     #endif
 
+    /**
+     * Use this method only if you already have your own IAP implementation using StoreKit 2 and want to use
+     * RevenueCat's backend. If you are using StoreKit 1 for your implementation, you do not need this method.
+     *
+     * You only need to use this method with *new* purchases. Subscription updates are observed automatically.
+     *
+     * #### Example:
+     *
+     * ```swift
+     * let result = try await product.purchase(options: options)
+     * Purchases.shared.handleObserverModeTransaction(purchaseResult: result)
+     * ```
+     *
+     * - Note: You need to finish the transaction yourself after calling this method.
```

Comment:
Maybe turn this into a `Warning` since it's pretty important? Maybe put that in the code example above too?